tests/test_queries.py

```diff
@@ -0,0 +1,54 @@
+import pytest
+import torch
+from perceiver_pytorch.queries import LearnableQuery
+from perceiver_pytorch.perceiver_io import PerceiverIO
+import einops
+
+
+@pytest.mark.parametrize("layer_shape", ["2d", "3d"])
+def test_learnable_query(layer_shape):
+    query_creator = LearnableQuery(
+        channel_dim=32,
+        query_shape=(6, 16, 16),
+        conv_layer=layer_shape,
+        max_frequency=64.0,
+        frequency_base=2.0,
+        num_frequency_bands=128,
+        sine_only=False,
+    )
+    x = torch.randn((4, 6, 12, 16, 16))
+    out = query_creator(x)
+    # Output is flattened, so should be [B, T*H*W, C]
+    assert out.shape == (4, 16 * 16 * 6, 803)
```

Comment:
Sorry, I probably just dont understand, where does the 803 come from?

Reply:
That is the number of channels of the `channel_dim` so 32 + the number of Fourier Features, which for this is 771

Reply:
And the number of Fourier Features comes from (num input axis)*((num_freq_bands*2)+1) so for this 3*(128*2+1) = 771

Reply:
Thanks for adding this